Doc/c-api/module.rst

```diff
@@ -451,8 +451,8 @@ state:
 
    On success, return ``0``. On error, raise an exception and return ``-1``.
 
-   Return ``NULL`` if *value* is ``NULL``. It must be called with an exception
-   raised in this case.
+   Return ``-1`` if *value* is ``NULL``. In this case, if the function is
+   called without an exception already set, it raises an exception.
```

Comment:
This is better, but I still find it slightly convoluted. BTW, we should document which exception will be raised.

How about something like this:

```suggestion
   If *value* is ``NULL`` and and exception is set, return ``-1``.
   If *value* is ``NULL`` and no exception is set,
   :exc:`SystemError` is raised.
```

Reply:
> `-1` is returned in both cases, so that would need to be factored in.

True. Perhaps the best candidate is the one proposed with commit 16c5efaf896eb27c836340c090122261c43eb84e